python/revng/model/v1/__init__.py

```diff
@@ -24,6 +29,62 @@ def get_reference_str(cls, t):
         return f"/TypeDefinitions/{t.ID}-{typename}"
 
 
+def make_diff(obj_old: StructBase, obj_new: StructBase) -> DiffSet:
+    return _make_diff(obj_old, obj_new, TypeHints, _generated.Binary)
+
+
+def get_type_info(path: str, root=_generated.Binary) -> StructBase:
+    return _get_type_info(path, root, TypeHints)
+
+
+def get_element_by_path(path: str, obj: StructBase) -> StructBase:
+    return _get_element_by_path(path, obj)
+
+
+def get_parent_element_by_path(path: str, obj: StructBase) -> StructBase:
+    return _get_parent_element_by_path(path, obj)
+
+
+def validate_diff(obj: StructBase, diff: DiffSet) -> bool:
+    return _validate_diff(obj, diff, get_type_info)
+
+
+def apply_diff(obj: StructBase, diff: DiffSet) -> tuple[bool, StructBase | None]:
+    return _apply_diff(obj, diff, validate_diff)
+
+
 # Since we subclassed them we need to re-register their constructors and representers
 YamlLoader.add_constructor("!Binary", Binary.yaml_constructor)
 YamlDumper.add_representer(Binary, Binary.yaml_representer)
+
+DiffYamlLoader.add_constructor("!DiffSet", DiffSet.yaml_constructor)
+DiffYamlDumper.add_representer(DiffSet, DiffSet.yaml_representer)
+
+
+def load_model(path: str) -> Binary:
+    with open(path) as f:
+        return yaml.load(f, Loader=YamlLoader)
+
+
+def load_model_from_string(model: str) -> Binary:
+    return yaml.load(model, Loader=YamlLoader)
+
+
+def dump_model(path: str, obj: StructBase) -> yaml:
+    with open(path, "w") as f:
+        yaml.dump(obj, f, Dumper=YamlDumper)
+
+
+def load_diff(path: str) -> DiffSet:
+    with open(path) as f:
+        diff = yaml.load(f, Loader=DiffYamlLoader)
+        return _parse_diff(diff, get_type_info)
+
+
+def dump_diff(path, diff) -> yaml:
+    with open(path, "w") as f:
+        yaml.dump(diff, f, Dumper=DiffYamlDumper)
+
+
+def dump_diff_to_string(diff) -> yaml:
+    return yaml.dump(diff, Dumper=DiffYamlDumper)
```

Comment:
Right, actually the return type is `None` so I will remove `yaml`. 

There are a few reasons I put them there:
1. They depend on the `TypeHint` dict which comes from the `_generated.py` which is imported in this file
2. Some function depend on the `_generated.Binary`  which comes from `_generated.py`
3. Some of them could be moved out of here, but I though it would be nice to expose functions that operate on the `module` and `diff` three so it's clear how to use the module

For example you can work with the module like so: `model.load("/path/to/model.yml")` and model.make_diff(model1, model2)`

Reply:
@mrjackv Can we turn these into factory class methods/regular methods?

Reply:
I'll check and report back

Reply:
Implemented as class methods